### PR TITLE
Update jQuery to v3.5.1.

### LINKED ISSUE #1014: Very old jQuery and jQueryUI Versions
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/AdminLTE/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/AdminLTE/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`6`

---
At the moment the following versions are used: 
- jQuery: 2.1.4 https://snyk.io/vuln/npm:jquery?lh=2.1.4
- jQueryUI: 1.11.3 https://snyk.io/vuln/npm:jquery-ui?lh=1.11.3

Dependency management can be very useful at this point. aka Greenkeeper with package.json

### PR DESCRIPTION
**Pull request by XhmikosR**
Signed-off-by: XhmikosR <xhmikosr@gmail.com>

**By submitting this pull request, I confirm the following:** `{please fill any appropriate checkboxes, e.g: [X]}`

`{Please ensure that your pull request is for the 'devel' branch!}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff))

---

I tested it as much as I could and it seems to work OK. Things that could break are one of the jQuery plugins, so we should test them out as good as possible.

It's not urgent to update, but we should try to do it ASAP. There have been several bugxifes, security fixes and plenty of performance improvements.

Browser support should be what we currently support, and I doubt anyone sane would use IE < 9 😛: https://jquery.com/upgrade-guide/3.0/#browser-support

I'm going to go through each of the breaking changes and see if we have code that's using that. 

https://jquery.com/upgrade-guide/3.0/

This would leave us 3rd-party code only.

TODO:

- [x] Test notify plugin
- [x] Test jquery confirm plugin
- [x] Test select plugin
- [x] Test toggle plugin
- [x] Address the `.context()` instances in our code
- [x] Address any newly added TODOs

Fixes #1014

### CODE REVIEW

File: scripts/pi-hole/js/auditlog.js [hunk 1]
```diff
@@ -134,12 +134,13 @@ function auditUrl(url) {
...
 
-$(document).ready(function () {
+$(function () {
   // Pull in data via AJAX
   updateTopLists();
 
   $("#domain-frequency tbody").on("click", "button", function () {
     var url = $(this).parents("tr")[0].textContent.split(" ")[0];
+    // TODO
     if ($(this).context.textContent === " Blacklist") {
```
Comment (XhmikosR) [inline scripts/pi-hole/js/auditlog.js]: This is deprecated, not sure what the original intention was?

Comment (PromoFaux) [reply scripts/pi-hole/js/auditlog.js]: Probably something that no longer matters, just remove it.

Comment (XhmikosR) [reply scripts/pi-hole/js/auditlog.js]: I'd like to wait for @DL6ER since it's probably his code. :)

**Comment by XhmikosR** [comment]
I added a TODO in the OP.

From my tests, this has a clear positive effect on load. We should definitely do it after we test the TODO. 🙂

File: scripts/pi-hole/js/auditlog.js (hunk 1 above)
Comment (XhmikosR) [reply scripts/pi-hole/js/auditlog.js]: It seems the whole click event is moot currently?

Comment (XhmikosR) [reply scripts/pi-hole/js/auditlog.js]: Actually, it is needed for the buttons to work. We could just change the target selectors, I think.

If that's the right approach, I can make another patch later.

Comment (XhmikosR) [reply scripts/pi-hole/js/auditlog.js]: @DL6ER friendly ping how to proceed with this. After I address this, only further testing is left.

**Comment by XhmikosR** [comment]
I'm going to split some of the patches that can land on develop later so that this branch is only left with the jQuery 3.x update.

**Comment by XhmikosR** [comment]
I'm gonna take a stab at fixing the TODO so that we can test this further, tops until tomorrow.

I'm not sure if we want to land this in v5.1.0, but the difference is pretty big IMO.

**Comment by PromoFaux** [comment]
> Test select plugin

Not sure what I'm looking for to test here, any pointers welcome

**Comment by XhmikosR** [comment]
Should be OK, I meant bootstrap-select. But all of the plugins are up to date so they should be fine.

The only plugin that's old is jquery-confirm. So if that works too, we should be fine.

EDIT: for what is worth that's one of the reasons I have it in my TODO (#1352) to remove jquery-confirm and use Bootstrap's alerts. But if it still works, it's not something we need to do for 5.1.0.

**Review by PromoFaux — APPROVED** [review]
Tested on all pages, no unexpected errors in console log. Not sure if further testing is required...

**Comment by XhmikosR** [comment]
We should be OK hopefully. Unless I missed another breaking change, but hey, mistakes can happen and we can fix them if reported.

**Comment by XhmikosR** [comment]
I mean the 3.x BC are a lot, so I tried to check every one of them in our code: https://jquery.com/upgrade-guide/3.0/

The plugins are all updated, so that's something too.

**Comment by yubiuser** [comment]
Not sure if it fits here, but there were some changes concerning "hidden" in 3.5.1.

I see now this:
Bildschirmfoto zu 2020-06-12 10-40-43

<br class='hidden-lg'>

**Comment by XhmikosR** [comment]
I already mentioned this somewhere else, but I don't think it's related to this change. Either way, please open new issues and don't comment on old ones because it's hard to track stuff.

**Comment by XhmikosR** [comment]
https://github.com/pi-hole/AdminLTE/pull/1161#issuecomment-640414058

**Comment by yubiuser** [comment]
https://github.com/pi-hole/AdminLTE/issues/1457

**Pull request merged**